### PR TITLE
Expose 'specified duration' variant of TOPP-RA in planner

### PR DESCRIPTION
**Pull request by vonHartz**
As discussed here: #91

### CODE REVIEW

File: mplib/planner.py [hunk 1]
```diff
@@ -378,6 +378,41 @@ def TOPP(self, path, step=0.1, verbose=False):
...
+        assert dof == len(self.joint_vel_limits)
+        assert dof == len(self.joint_acc_limits)
+        ss = np.linspace(0, 1, N_samples)
+        path = ta.SplineInterpolator(ss, path)
+        pc_vel = constraint.JointVelocityConstraint(self.joint_vel_limits)
+        pc_acc = constraint.JointAccelerationConstraint(self.joint_acc_limits)
+        instance = algo.TOPPRAsd(
+            [pc_vel, pc_acc], path, parametrizer="ParametrizeConstAccel"
+        )
+        instance.set_desired_duration(duration)
```
Comment (Lexseal) [inline mplib/planner.py:402]: seems to me this is the only change. would you mind just modifying the TOPP function above and add `duration=None` as default argument?

**Review by Lexseal — CHANGES_REQUESTED** [review]
too much code duplication. can be done with default duration argument + conditional

File: mplib/planner.py (hunk 1 above)
Comment (vonHartz) [reply mplib/planner.py:402]: Yes, that's correct.
Though you might want to know that TOPPRAsd is  a bit less efficient than TOPRRA.
See: https://github.com/hungpham2511/toppra/blob/730aa8ad6854c7da01304ad1ed8c0819400862f4/toppra/algorithm/reachabilitybased/desired_duration_algorithm.py#L20
If that is not a concern for you, I'm happy to make the change.

Comment (vonHartz) [reply mplib/planner.py:402]: Ah, nevermind. I'll do it with a conditional.

**Comment by vonHartz** [comment]
Made the requested change. Let me know if there's anything else.

**Review by Lexseal — APPROVED** [review]
LGTM

**Review by KolinGuo — APPROVED** [review]

**Comment by KolinGuo** [comment]
Thanks for your contribution @vonHartz

**Comment by KolinGuo** [comment]
Please use the nightly wheels if you need to use it for now. A new minor version release is planned in the coming weeks.

**Pull request merged**